cloud/migrate-to-cloud/hypershift.md

```diff
@@ -0,0 +1,172 @@
+---
+title: Migrate with Hypershift
+excerpt: Migrate an existing PostgreSQL database to Timescale Cloud in a single step
+product: cloud
+keywords: [data migration, Hypershift]
+tags: [ingest, Hypershift, postgresql]
+---
+
+import Experimental from 'versionContent/_partials/_experimental.mdx';
+
+# Migrate with Hypershift
+
+You can use Hypershift&nbsp;0.2&nbsp;(beta) to migrate existing PostgreSQL
+databases in one step, and enable compression and hypertable creation on the
+fly.
+
+Hypershift can migrate your data in Timescale Cloud from these sources:
+
+*   Standard PostgreSQL databases
+*   Amazon RDS databases
+*   Other TimescaleDB databases, including Managed Service for Timescale
+
+Because compression is enabled during the migration, you do not need to have the
+maximum amount of storage available in the target database before you start
+migration.
+
+<Experimental />
+
+In preliminary testing, Hypershift migrated 60&nbsp;GB of data in 9 to 12 minutes,
+and 1&nbsp;TB of data in under 4 hours. You can continue reading from your
+source database during this time, though performance could be slower. If you
+write to tables in your source database during the migration, the new writes
+might not be transferred to Timescale Cloud. To avoid this problem, fork your
+database and migrate your data from the fork. To avoid this problem, see the
+section on [migrating an active database]
+(<http://docs.timescale.com/cloud/latest/migrate-to-cloud/#migrate-an-active-database>).
+
+<highlight type="important">
+If you have a very large database, and Hypershift is going to have to run for a
```

Comment:
Another important point that is associated with large/busy databases is that Hypershift will hold a transaction open for the entire duration of the migration. This will prevent autovacuum from running, which could cause issues ranging from poor performance to transaction wraparound. It might be worthwhile to mention that here.

Reply:
Added it in. Do we have any specific troubleshooting info for this yet?

Reply:
It's kind of an advanced problem, so I'm not sure that we have materials for it. I suspect that if people would potentially run into these kinds of issues, they are aware of it, so just mentioning it is enough.